Private/gokay/impress preview refresh

gokaysatir opened the pull request:
* Resolves: # <!-- related github issue -->
* Target version: master 

### Summary


### TODO

- [ ] ...

### Checklist

- [ ] I have run `make prettier-write` and formatted the code.
- [ ] All commits have Change-Id
- [ ] I have run tests with `make check`
- [ ] I have issued `make run` and manually verified that everything looks okay
- [ ] Documentation (manuals or wiki) has been updated or is not required

caolanm commented:
The first two commits look good. I'm not sure about the 3rd, I think I'd like to see how to reproduce that one myself. (There is a conflict reported, so I think  a rebase is needed anyay)

caolanm commented:
I think the third patch is working around a (very odd) problem in core impress that switching from notes to standard doesn't actually change the selected page so there ends up previews getting rendered in "standard" mode that actually render the still selected "note" page so note previews can show up pretending to be standard previews. I believe  https://gerrit.libreoffice.org/c/core/+/176084 should fix that. If that works, then we could drop this 3rd patch and the remainder is nice-to-have tidy ups.

gokaysatir commented:
> I think the third patch is working around a (very odd) problem in core impress that switching from notes to standard doesn't actually change the selected page so there ends up previews getting rendered in "standard" mode that actually render the still selected "note" page so note previews can show up pretending to be standard previews. I believe https://gerrit.libreoffice.org/c/core/+/176084 should fix that. If that works, then we could drop this 3rd patch and the remainder is nice-to-have tidy ups.

Rebased and dropped the 3rd commit. Thanks for the help :)
